File: API/DTOs/MemberDto.cs
using System.ComponentModel.DataAnnotations;
using API.Entities;

namespace API.DTOs
{
    public class MemberDto : BaseEntity
    {
        [Required]
        [DataType(DataType.EmailAddress)]
        [EmailAddress]
        public string UserName { get; set;}
        public string FirstName{get;set;}
        public string LastName{get;set;}
        public DateOnly DateOfBirth { get; set; }
        public Adress Adress{ get; set; }
        public string Role { get; set; }
        public List<ShoppingCart> Orders { get; set;} 
        public List<Voucher> Vouchers{get;set;}
    }
}

File: client/src/app/_models/memberDto.ts
import { Adress } from "./adress";
import { Order } from "./order";
import { Voucher } from "./voucher";

export interface MemberDto{
    id:number,
    userName: string,
    firstName: string,
    lastName:string,
    password: string,
    role: string,
    orders:Order[],
    vouchers: Voucher[],
    adress:Adress, 
    dateOfBirth: any
}